src/controllers/krate/downloads.rs

```diff
@@ -75,10 +99,74 @@ pub async fn get_crate_downloads(state: AppState, path: CratePath) -> AppResult<
         downloads: i64,
     }
 
+    let include = params
+        .include
+        .as_ref()
+        .map(|mode| ShowIncludeMode::from_str(mode))
+        .transpose()?
+        .unwrap_or_default();
+
+    if include.versions {
+        latest_five.sort_unstable_by_key(|it| cmp::Reverse(it.id));
```

Comment:
I assume this sort operation is the reason for the `split_at_mut()` change above? is there a reason why we need to re-sort here? can't we use `latest_five` in the reverse semver order that they should already be in?

Reply:
`VersionOwnerAction::for_versions` is using `.grouped_by(versions)` internally to accomplish that. seems less brittle to me 🤔

Reply:
I'm not sure I understand the feedback. Do you mean we should encapsulate this in models, or share a hashmap for publishers and owner actions?

---

I also noticed that we should probably query minimal versions data instead of full columns for downloads.

Reply:
I'm talking about https://docs.rs/diesel/latest/diesel/associations/trait.GroupedBy.html. If we used that for the `id_and_publishers` query below too then we wouldn't need the sorting and can guarantee that the mapping is correct, even if for some unlikely reason one of the ids couldn't be found.

Reply:
Rebased and rewrote with `grouped_by(latest_five)`.